src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -1120,19 +1122,51 @@ void SpatialPooler::updateDutyCyclesHelper_(vector<Real>& dutyCycles,
 
 void SpatialPooler::updateBoostFactors_()
 {
-  for (UInt i = 0; i < numColumns_; i++)
+
+  if (maxBoost_ > 1.0)
   {
-    if (minActiveDutyCycles_[i] <= 0)
+    vector<Real> targetDensity(numColumns_, 0);
+    if (globalInhibition_)
     {
-      continue;
+      Real density = localAreaDensity_;
```

Comment:
I think it'd be more readable (and infinitesimally faster) to handle the two cases in if/else blocks.

~~~c++
Real density;
if (numActiveColumnsPerInhArea_ > 0)
{
  // ...
}
else
{
  density = localAreaDensity_;
}
~~~